### PR TITLE
Import Everything to 2021

### PR DESCRIPTION
**Pull request by KenwoodFox**
This major pull request should be completed once ALL of last year's code is ported into the 2021 robot.

### CODE REVIEW

File: src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -42,16 +42,16 @@
 public class RobotContainer {
   // Joysticks and Operator Input
   public static final Joystick DriverStick = new Joystick(Constants.DriverInputSettings.Driver_Stick_Port);
-  public static final Joystick OperatorStick = new Joystick(Constants.OperatorInputSettings.Operator_Controller_Port);
-  public static final Joystick DSTogglePanel = new Joystick(Constants.DSTogglePanelSettings.DS_Toggle_Panel_Port);
+  //public static final Joystick OperatorStick = new Joystick(Constants.OperatorInputSettings.Operator_Controller_Port);
```
Comment (KenwoodFox) [inline src/main/java/frc/robot/RobotContainer.java:45]: We'll need to add this back

File: src/main/java/frc/robot/RobotContainer.java [hunk 2]
```diff
@@ -42,16 +42,16 @@
...
+  //public static final Joystick DSTogglePanel = new Joystick(Constants.DSTogglePanelSettings.DS_Toggle_Panel_Port);
```
Comment (KenwoodFox) [inline src/main/java/frc/robot/RobotContainer.java:46]: We'll need to add this back

File: src/main/java/frc/robot/RobotContainer.java [hunk 3]
```diff
@@ -42,16 +42,16 @@
...
-  private final Intake intake = new Intake();
-  private final ROS ros = new ROS();
-  private final Shooter shooter = new Shooter();
-  private final Climber climber = new Climber();
-  private final Solver solver = new Solver();
+  //private final Intake intake = new Intake();
+  //private final ROS ros = new ROS();
+  //private final Shooter shooter = new Shooter();
+  //private final Climber climber = new Climber();
+  //private final Solver solver = new Solver();
```
Comment (KenwoodFox) [inline src/main/java/frc/robot/RobotContainer.java:54]: We'll need to add this back

File: src/main/java/frc/robot/RobotContainer.java [hunk 4]
```diff
@@ -76,28 +76,28 @@ public RobotContainer() {
     SmartDashboard.putData("Handling Mode", handlingChooser);
 
     // Autonomous Mode
-    autoChooser.setDefaultOption("ROS Full Auto", new ROSControl(drivetrain, ros, shooter));
+    //autoChooser.setDefaultOption("ROS Full Auto", new ROSControl(drivetrain, ros, shooter));
```
Comment (KenwoodFox) [inline src/main/java/frc/robot/RobotContainer.java:79]: We'll need to add this back

File: src/main/java/frc/robot/RobotContainer.java [hunk 5]
```diff
@@ -76,28 +76,28 @@ public RobotContainer() {
...
     autoChooser.addOption("Do Nothing", null); // Send null
     SmartDashboard.putData("Auto Mode", autoChooser);
 
     // NT functions
     SmartDashboard.putData("Reset Encoders", new ResetEncoders(drivetrain));
     SmartDashboard.putData("Shift Up", new ShiftUp(drivetrain)); // For testing only!
     SmartDashboard.putData("Shift Down", new ShiftDown(drivetrain)); // For testing only!
-    SmartDashboard.putData("Arm Shooter", new ArmShooter(shooter)); // For testing only!
+    //SmartDashboard.putData("Arm Shooter", new ArmShooter(shooter)); // For testing only!
```
Comment (KenwoodFox) [inline src/main/java/frc/robot/RobotContainer.java:87]: We'll need to add this back

File: src/main/java/frc/robot/RobotContainer.java [hunk 6]
```diff
@@ -76,28 +76,28 @@ public RobotContainer() {
...
-    shooter.setDefaultCommand(new ManualShooter(shooter, OperatorStick, () -> OperatorStick.getRawAxis(3), () -> (OperatorStick.getRawAxis(Constants.OperatorInputSettings.Turret_Spin_cw_axis) - OperatorStick.getRawAxis(Constants.OperatorInputSettings.Turret_Spin_ccw_axis))));
-    climber.setDefaultCommand(new ManualClimb(climber, () -> OperatorStick.getRawAxis(Constants.OperatorInputSettings.Gantry_Axis), () -> OperatorStick.getRawAxis(Constants.OperatorInputSettings.Climb_Axis)));
+    //shooter.setDefaultCommand(new ManualShooter(shooter, OperatorStick, () -> OperatorStick.getRawAxis(3), () -> (OperatorStick.getRawAxis(Constants.OperatorInputSettings.Turret_Spin_cw_axis) - OperatorStick.getRawAxis(Constants.OperatorInputSettings.Turret_Spin_ccw_axis))));
+    //climber.setDefaultCommand(new ManualClimb(climber, () -> OperatorStick.getRawAxis(Constants.OperatorInputSettings.Gantry_Axis), () -> OperatorStick.getRawAxis(Constants.OperatorInputSettings.Climb_Axis)));
```
Comment (KenwoodFox) [inline src/main/java/frc/robot/RobotContainer.java:95]: We'll need to add this back

File: src/main/java/frc/robot/RobotContainer.java [hunk 7]
```diff
@@ -76,28 +76,28 @@ public RobotContainer() {
...
 
     // ROS Commands
-    ros.publishCommand("resetEncoders", new ResetEncoders(drivetrain));
-    ros.publishCommand("shiftUp", new ShiftUp(drivetrain));
-    ros.publishCommand("shiftDown", new ShiftDown(drivetrain));
+    //ros.publishCommand("resetEncoders", new ResetEncoders(drivetrain));
+    //ros.publishCommand("shiftUp", new ShiftUp(drivetrain));
+    //ros.publishCommand("shiftDown", new ShiftDown(drivetrain));
```
Comment (KenwoodFox) [inline src/main/java/frc/robot/RobotContainer.java:100]: We'll need to add this back

File: src/main/java/frc/robot/RobotContainer.java [hunk 8]
```diff
@@ -107,14 +107,14 @@ public RobotContainer() {
    * {@link edu.wpi.first.wpilibj2.command.button.JoystickButton}.
    */
   private void configureButtonBindings() {
-    new JoystickButton(DriverStick, Constants.DriverInputSettings.Autonomous_Restart_Button).whenPressed(new ROSControl(drivetrain, ros, shooter)); // When you press the Autonomous Restart Button
+    //new JoystickButton(DriverStick, Constants.DriverInputSettings.Autonomous_Restart_Button).whenPressed(new ROSControl(drivetrain, ros, shooter)); // When you press the Autonomous Restart Button
```
Comment (KenwoodFox) [inline src/main/java/frc/robot/RobotContainer.java:110]: We'll need to add this back

File: src/main/java/frc/robot/RobotContainer.java [hunk 9]
```diff
@@ -107,14 +107,14 @@ public RobotContainer() {
...
-    new JoystickButton(OperatorStick, Constants.OperatorInputSettings.Disarm_Shooter_Button).whenPressed(new DisarmShooter(shooter));
-    new JoystickButton(OperatorStick, Constants.OperatorInputSettings.Intake_Button).whenHeld(new SpinIntake(intake));
-    new JoystickButton(OperatorStick, Constants.OperatorInputSettings.Purge_Button).whenHeld(new PurgeIntake(intake));
-    new JoystickButton(DSTogglePanel, Constants.DSTogglePanelSettings.SolveStageTwo).whenPressed(new SolveStage2(solver));
+    //new JoystickButton(OperatorStick, Constants.OperatorInputSettings.Arm_Shooter_Button).whenPressed(new ArmShooter(shooter)); // Arms and disarms the shooter
+    //new JoystickButton(OperatorStick, Constants.OperatorInputSettings.Disarm_Shooter_Button).whenPressed(new DisarmShooter(shooter));
+    //new JoystickButton(OperatorStick, Constants.OperatorInputSettings.Intake_Button).whenHeld(new SpinIntake(intake));
+    //new JoystickButton(OperatorStick, Constants.OperatorInputSettings.Purge_Button).whenHeld(new PurgeIntake(intake));
+    //new JoystickButton(DSTogglePanel, Constants.DSTogglePanelSettings.SolveStageTwo).whenPressed(new SolveStage2(solver));
```
Comment (KenwoodFox) [inline src/main/java/frc/robot/RobotContainer.java:117]: We'll need to add this back

File: src/main/java/frc/robot/commands/functions/ResetEncoders.java [hunk 10]
```diff
@@ -46,5 +45,5 @@ public boolean isFinished() {
     }else{
     return false;
     }
-  }
+  } //TODO: Clean up the code
```
Comment (KenwoodFox) [inline src/main/java/frc/robot/commands/functions/ResetEncoders.java:48]: Don't forget this TODO

**Review by KenwoodFox — COMMENTED** [review]
This looks good to set as master, even if every feature is not fully imported.

File: src/main/java/frc/robot/RobotContainer.java (hunk 3 above)
Comment (OneBiteAidan) [inline src/main/java/frc/robot/RobotContainer.java:54]: ROS Goodness needs to be re-implemented Soon™

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 11]
```diff
@@ -13,97 +12,40 @@
...
-    portMotorSlave0.setInverted(Constants.DrivetrainPID.portMotorInvert);
-    portMotorSlave1.setInverted(Constants.DrivetrainPID.portMotorInvert);
-
-    starboardMotor.setInverted(Constants.DrivetrainPID.starboardMotorInvert);
-    starboardMotorSlave0.setInverted(Constants.DrivetrainPID.starboardMotorInvert);
-    starboardMotorSlave1.setInverted(Constants.DrivetrainPID.starboardMotorInvert);
+    portMotor.setInverted(false);
+    starboardMotor.setInverted(true);
+    //portMotor.setInverted(Constants.DrivetrainPID.portMotorInvert); // Sets the output of the motor backwards
+    //starboardMotor.setInverted(Constants.DrivetrainPID.starboardMotorInvert);
```
Comment (OneBiteAidan) [inline src/main/java/frc/robot/subsystems/Drivetrain.java:41]: We should change the inversion point of the drivetrain to one place. There's also inversion in HumanControl.java

File: src/main/java/frc/robot/RobotContainer.java (hunk 3 above)
Comment (KenwoodFox) [reply src/main/java/frc/robot/RobotContainer.java:54]: It will be done Soon™

File: src/main/java/frc/robot/subsystems/Drivetrain.java (hunk 11 above)
Comment (KenwoodFox) [reply src/main/java/frc/robot/subsystems/Drivetrain.java:41]: Good point, ill make a new branch for `driver-cleanup` to fix this in the future.

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 12]
```diff
@@ -114,8 +56,21 @@ public Drivetrain() {
    * @param thro
    */
   public void FlyByWireA(double steerage, double thro){
-    starboardMotor.set(ControlMode.PercentOutput, (thro + (steerage)) * -1); // Set the starboard motor to the sum of thro - steerage
-    portMotor.set(ControlMode.PercentOutput, (thro - (steerage)) * -1); // Set the port motor to the sum of thro + steerage
+    starboardMotor.set(thro + (steerage)); // Set the starboard motor to the sum of thro - steerage
+    portMotor.set(thro - (steerage)); // Set the port motor to the sum of thro + steerage
+  }
+
```
Comment (OneBiteAidan) [inline src/main/java/frc/robot/subsystems/Drivetrain.java:62]: We had removed some inversions here. They should be done solely in lines 37 and 38 of drivetrain.java

**Review by OneBiteAidan — APPROVED** [review]

File: src/main/java/frc/robot/subsystems/Drivetrain.java (hunk 12 above)
Comment (KenwoodFox) [reply src/main/java/frc/robot/subsystems/Drivetrain.java:62]: Complicated inversions will be resolved in `driver-cleanup`

**Pull request merged**